moto/awslambda/policy.py

```diff
@@ -78,12 +74,17 @@ def del_statement(self, sid: str, revision: str = "") -> None:
     # converts AddPermission request to PolicyStatement
     # https://docs.aws.amazon.com/lambda/latest/dg/API_AddPermission.html
     def decode_policy(self, obj: Dict[str, Any]) -> "Policy":
-        policy = Policy(self.parent)
+        policy = Policy(self.parent)  # type: ignore[no-untyped-call]
         policy.revision = obj.get("RevisionId", "")
+        # get function_arn or arn from parent
+        if hasattr(self.parent, "arn"):
```

Comment:
If we make the typing explicit, we can then use that info to clarify this if/else statement:
`if isinstance(self.parent, LayerVersion):`

That makes it a lot more readable for future developers.